src/gc.c

```diff
@@ -2557,10 +2547,8 @@ FORCE_INLINE void gc_mark_outrefs(jl_ptls_t ptls, jl_gc_markqueue_t *mq, void *_
                 if (new_obj != NULL) {
                     if (!meta_updated)
                         goto mark_obj;
-                    else {
-                        assert_validity_of_new_obj(parent, new_obj);
+                    else
                         gc_ptr_queue_push(mq, new_obj);
-                    }
```

Comment:
Why were you able to delete all of these? Is it because you moved the assertion to happen before this whole function is called?

Reply:
The `gc_assert_parent_validity` in `gc_mark_obj{8, 16, 32}` should already cover these cases.